crates/bevy_asset/src/path.rs

```diff
@@ -180,10 +180,25 @@ impl<'a> From<&'a Path> for AssetPath<'a> {
 }
 
 impl<'a> From<PathBuf> for AssetPath<'a> {
-    fn from(path: PathBuf) -> Self {
-        AssetPath {
-            path: Cow::Owned(path),
-            label: None,
+    fn from(mut path: PathBuf) -> Self {
+        match path.file_name().map(ToOwned::to_owned) {
+            Some(os_str) => {
+                let mut parts = os_str
+                    .to_str()
+                    .expect("File name format is not valid unicode")
```

Comment:
Generally you shouldn't panic inside a `From` impl. See [this](https://rust-lang.github.io/rust-clippy/master/index.html#fallible_impl_from).

You should probably handle this case and return the `None` branch as well.